### ISSUE TITLE
App Overview and Use Cases

### ISSUE THREAD

**Issue by tvledesign** [cmojarro/FantasticFour#2]
APPLICATION OVERVIEW

“Homework Helper” (temporary name) is a local application that can be install via the user’s mobile device or their computer’s browser. What Homework Helper does is essential a “To Do List” application with a little more aim towards high school and college student who are trying to keep track of their assignments and prioritize between each one as well as keep track when they are expected to be due.

---

APPLICATION FEATURE

Basic features of the app includes:
- Basic “To Do List” feature that allows the user to create a list of tasks.
- Sub-Lists for any sub tasks that needs to be completed before a task is marked done.
- Color coding for prioritizing between each tasks.
- Expected and due dates to set goals for when the user want the task finish and when it is actually due.
- Notification reminder on tasks for friendly reminders to tell the user when a task is due. 

---

USE CASES - GENERAL USERS

Users who will be using the app are mainly targeted towards students, mainly:
- High School Students
- College Students

Users who can fall within the app’s target audience can include:
- Interns
- Project Managers or in general Managers
- Workers

---

USE CASES - AVERAGE USAGE 
- Add Task
- Add Sub-Task
- Mark Task Complete
- Mark Sub-Task Complete
- Setup Notification Reminder
- Setup Expected Due Date
- Setup Actual Due Date
- Prioritize Task

---

USE CASES - INDEX

Use Case ID Use Case Name   User    Complexity
1   High School High School Student 2 out of 5
1   College College Student 4 out of 5

---

USE CASES - HIGH SCHOOL STUDENT

Use Case Number: 1

Use Case Name: High School 

User Case User: High School Student

Description: A high school student finds herself sometimes forgetting to write down her homework from time to time and she normally has a hard time recalling what she had to do throughout the day. Seeing that she has a different homework assignment everyday it can be very bothersome trying to keep track of things that normally changes. With the option of using Homework Helper as an alternative she downloads the app onto her mobile device and at the end of each class does the following:
1. Launches Homework Helper
2. Creates a new task for each homework assignment due by pressing the add task icon. 

Ex. After Geometry class ended, the homework assigned was a worksheet she had to finish by tomorrow. A basic scenario would be to: open the application, click on the add button to add a new task and type in “Finish Geometry Worksheet.”

Now if she was assigned a project however the following would still apply except:
1. Assign an Expected Due Date for when she wants to complete the tasks by.
2. Assign the Actual Due Date to keep track of when it’s due.

Additional options would be to:
1. Setup Notification Reminder to get a daily reminder of the tasks.
2. When finished check the box next to the task to mark as complete.

Note: The only thing to keep in mind here is a high school student’s homework changes daily so some feature may not always be used such as daily reminder, color coding, due dates, and sub-tasks.

---

USE CASES - COLLEGE STUDENT

Use Case Number: 2

Use Case Name: College

User Case User: College Student

Description: College students generally have a lot of homework they have to keep track of but unlike high school students, they tend to have more time to do their homework because classes can be anywhere from once to twice a week. A typical usage for a college student would be:
1. Launches Homework Helper
2. Creates a new task for each homework assignment due by pressing the add task icon.

OR
1. Launches Homework Helper
2. Create a task for each class the student has by pressing the add task icon.

The biggest difference is that a college student will have more than one assignment per class so instead of making a task each assignment a typical student can make a task per class and:
1. For each class task, create a sub-list full of homework assignments each class.
2. Set up the Expected Due Date on when to finish the class’s homework assignment.
3. Set up the Actual Due Date which can be the date the next class starts.
4. Prioritize based on which class the student will have first.

Note: Since a college student has a much bigger work load it is possible they will make use of all the features on the app list and use it differently than an average student.

**Comment by tvledesign** [comment]
General Use Cases for two average users whom are college and high school students. I also posted some application overviews and general information on the application itself.
